### PR TITLE
ci: run storybook pipeline on html package changes

### PR DESCRIPTION
**Pull request by hamza14khan**
- Verify if `govie-frontend/storybook` package was affected by the associated dependencies and make appropriate deployment.
- Also, enable manual deployment.

### CODE REVIEW

File: .github/workflows/deploy-html-storybook-prod.yml [hunk 1]
```diff
@@ -4,6 +4,8 @@ on:
   push:
     branches:
       - main
+    paths:
+      - "packages/html/**"
```
Comment (ianlnf) [inline .github/workflows/deploy-html-storybook-prod.yml:8]: The issue with this approach is that any changes in the downstream dependencies of the html packages will not trigger a new Storybook deployment, e.g. tokens package, theme package etc. You'd be better to use the `nx affected` command and add an Nx deploy script for each storybook instance that programatically pushes to S3.

Comment (hamza14khan) [reply .github/workflows/deploy-html-storybook-prod.yml:8]: Good catch! Let me fix this.

File: .github/workflows/deploy-html-storybook-prod.yml [hunk 2]
```diff
@@ -4,26 +4,73 @@ on:
...
+
+      - name: Install dependencies
+        run: |
+          corepack enable
+          pnpm install --frozen-lockfile --ignore-scripts
+
+      - name: Determine affected projects
+        id: affected
+        run: |
+          npx nx affected:libs --base=origin/main~1 --head=HEAD > affected.txt
```
Comment (pregno) [inline .github/workflows/deploy-html-storybook-prod.yml:39]: This command in local doesn't work, does `affected:libs` need to be present in package.json?

**Review by ianlnf — APPROVED** [review]

File: .github/workflows/deploy-html-storybook-prod.yml [hunk 3]
```diff
@@ -4,26 +4,72 @@ on:
...
+
+      - name: Install dependencies
+        run: |
+          corepack enable
+          pnpm install --frozen-lockfile --ignore-scripts
+
+      - name: Determine affected projects
+        id: affected
+        run: |
+          nx show projects --affected --base=origin/main~1 --head=HEAD > affected.txt cat affected.txt
```
Comment (pregno) [inline .github/workflows/deploy-html-storybook-prod.yml:39]: Is `cat affected` needed?
Shouldn't it be on a new line?

**Review by pregno — APPROVED** [review]

**Pull request merged**